src/client.rs

```diff
@@ -45,34 +45,47 @@ impl Client {
     }
 
     /// Returns an async connection from the client.
-    pub fn get_async_connection(
-        &self,
-    ) -> impl Future<Item = crate::aio::Connection, Error = RedisError> {
-        crate::aio::connect(self.connection_info.clone())
+    pub fn get_async_connection<'a>(
+        &'a self,
+    ) -> impl Future<Output = RedisResult<crate::aio::Connection>> + 'a {
+        crate::aio::connect(&self.connection_info)
     }
 
     /// Returns a async shared connection from the client.
     ///
     /// This uses the default tokio executor.
     #[cfg(feature = "executor")]
-    pub fn get_shared_async_connection(
-        &self,
-    ) -> impl Future<Item = crate::aio::SharedConnection, Error = RedisError> {
-        let executor = tokio_executor::DefaultExecutor::current();
-        self.get_async_connection()
-            .and_then(move |con| crate::aio::SharedConnection::new(con, executor))
+    pub fn get_shared_async_connection<'a>(
+        &'a self,
+    ) -> impl Future<Output = RedisResult<crate::aio::SharedConnection>> + 'a {
+        struct TokioExecutor;
```

Comment:
Personally I find it pretty unclear what this is trying to achieve. Maybe add some comments?